### PR TITLE
feat: tracked response statuscodes

### LINKED ISSUE #200: Provide capability to exclude API endpoints for request tracking
**Is your feature request related to a problem? Please describe.**
Provide the capability to exclude API endpoints for request tracking since they might not be relevant, or you want to optimize the cost for all your telemetry.

In some cases; you'd like to filter all requests while for others you might want to filter them out based on the status code.

**Describe the solution you'd like**
I'd love to have flexibility in the sense that:
- I can apply an attribute on my operation to opt-out for tracking
  - Would be nice to have optional flags there to give granularity to only opt-out for request, response or both
  - Provide an optional flag to define what status codes should be tracked, all that are not specified are considered to be excluded
- Same as above, but through the request logging options instead

That said, I personally, would be fine if we start with attributes only since describing all of this through the options would be either hard or verbose and not really what we would use I think

### PR DESCRIPTION
**Pull request by stijnmoreels**
Provides an additional request tracking option that describes what response status codes should be tracked.
Provides an additional input for the request tracking options attribute that describes the same thing.

Closes #200

### CODE REVIEW

**Comment by tomkerkhove** [comment]
In this case we are more using an opt-in instead of an opt-out, no? Ideally I'd be able to say "don't track 200 OK, but do track all the rest!"

**Comment by stijnmoreels** [comment]
> In this case we are more using an opt-in instead of an opt-out, no? Ideally I'd be able to say "don't track 200 OK, but do track all the rest!"

Yes, this is an opt-in as described in the issue: _"Provide an optional flag to define what status codes should be tracked, all that are not specified are considered to be excluded"_.

**Comment by tomkerkhove** [comment]
Hm, I've misformulated it then but that's ok.

**Comment by tomkerkhove** [comment]
One thing I wwould add then is support for ranges so that I can include everything in the 500 or 500-505 range.

The reason why we need this, for example, is to reduce the telemetry that our health endpoint is generating so that we only have the 5xx responses.

File: docs/preview/features/logging.md [hunk 1]
```diff
@@ -255,6 +257,75 @@ So the resulting log message becomes:
...
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+public class Startup
+{
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseRequestTracking(options => options.TrackedStatusCodes.Add(HttpStatusCode.Created));
```
Comment (tomkerkhove) [inline docs/preview/features/logging.md:296]: We are using Dapr which add a probe endpoint that we don't control.

In order to filter that out, we would maybe need to define this here somehow based on the path.
How hard would it be to add that?

Comment (stijnmoreels) [reply docs/preview/features/logging.md:296]: That's more something for another issue and PR, but I'm thinking about routes or route ranges (wildcards) that you could add to the options if you can't control it yourself.
In other scenario's, you could use our `ExcludeRequestTracking` attribute I guess.

Comment (tomkerkhove) [reply docs/preview/features/logging.md:296]: It's not really the same as we want to limit on a set of status codes, not request/response payload.

Comment (tomkerkhove) [reply docs/preview/features/logging.md:296]: Seperate task for this is ok though.

File: docs/preview/features/logging.md [hunk 2]
```diff
@@ -255,6 +257,75 @@ So the resulting log message becomes:
 
 `HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [X-Api-Key,])`
 
+#### Tracked HTTP status codes
```
Comment (tomkerkhove) [inline docs/preview/features/logging.md:260]:
```suggestion
#### Reducing requests to specific HTTP status code(s)
```

File: docs/preview/features/logging.md [hunk 3]
```diff
@@ -255,6 +257,75 @@ So the resulting log message becomes:
...
+More grainer control can be achieved via placing an attribute on either the controller's class definition or the endpoint method:
+
+```csharp
+using Arcus.WebApi.Logging;
+
+[ApiController]
+public class OrderController : ControllerBase
+{
+    [HttpPost]
+    [RequestTracking(HttpStatusCode.Created)]
```
Comment (tomkerkhove) [inline docs/preview/features/logging.md:311]: Would love to support this:
```suggestion
    [RequestTracking(HttpStatusCode.Created, HttpStatusCode.OK)]
```

Or define a range from 200 -> 299.

File: docs/preview/features/logging.md [hunk 4]
```diff
@@ -168,6 +168,8 @@ public class OrderController : ControllerBase
 }
 ```
 
+#### Excluding request/response bodies
```
Comment (tomkerkhove) [inline docs/preview/features/logging.md:171]: Do we need to update a TOC?

File: docs/preview/features/logging.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/preview/features/logging.md:311]: Ah, that would be better. Now it's supported for multiple ones that you can inside one attribute but we could use it indeed for ranges. That would be better maybe.

Comment (stijnmoreels) [reply docs/preview/features/logging.md:311]: But let's do that in another issue too. Now it already supports this syntax but it takes the two status codes separatly in. Maybe for ranges, we should come up with our own enumeration or let it specify as integers.

File: docs/preview/features/logging.md (hunk 4 above)
Comment (stijnmoreels) [reply docs/preview/features/logging.md:171]: No, as the TOC only contains the main headers.

File: docs/preview/features/logging.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/preview/features/logging.md:311]: I'll remove the `params` definition for now so we can add these ranges later.

Comment (stijnmoreels) [reply docs/preview/features/logging.md:311]: Ok, removed these `params`. I'll create an issue for the ranges.

Comment (stijnmoreels) [reply docs/preview/features/logging.md:311]: Done https://github.com/arcus-azure/arcus.webapi/issues/216, resolving.

File: docs/preview/features/logging.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/features/logging.md:296]: Ok, not entirely sure what the goal is here, so I suggest that you make the additional issue for this one. 😄 I'll resolve this comment.

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**